### PR TITLE
Added basic functionality to UI

### LINKED ISSUE #76: Add functionality to UI
The UI is currently taking user input, but it isn't actually producing a cloned output.

I think the easiest (maybe not the best) route is to call ```main.py```  when the "Clone voice" button is clicked.

I think you can do this through the ```sys``` library. Read ```stdout``` to display success or errors on the UI.

### PR DESCRIPTION
**Pull request by andrew-fennell**
## Summary
Added functionality to the "Clone voice" button. It now runs ```python -m CogNative.main``` with all arguments given.

The final line of the output will be printed to the GUI. This is typically "ERROR: <some error>" or "Cloned output to <some location>". It will either print in red or green, depending on if it is an "ERROR".

The way I have done everything in this branch is NOT the best way to do it. I am trying to get the UI to a workable place so it can be improved upon.

**I changed the saved_embeds path to just be their filename + .ckpt. This makes it easier to use with the UI.**

### Success case
image

### Error Case
image

All inputs on the UI should be functional. I removed any that would no longer be used (with language auto-detection added in #78 and #71)

Closes #76

### CODE REVIEW

File: CogNative/testUI/UI.py [hunk 1]
```diff
@@ -63,31 +60,108 @@ def displayProgressBar():
...
-        )
-    # If it hasn't set directory to file name
+    output_dir = data["output"]["output_dir"]
+    data["output"]["output_audio_path"] = f"{output_dir}/{output_name_entry.get()}"
+    output_dir_lbl.config(text=data["output"]["output_audio_path"])
+
+    cmd = ['python', '-m', 'CogNative.main',
+           '-sampleAudio', data["input"]["audio_to_clone"],
+           '-out', data["output"]["output_audio_path"],
+           '-useExistingEmbed', 'y'
```
Comment (JacobSmith1020) [inline CogNative/testUI/UI.py]: This should be connected to the corresponding check box.

**Review by JacobSmith1020 — CHANGES_REQUESTED** [review]
Suggestions: Might want to show what folders the user selected for the various audio inputs in the UI. We also might want to make it more clear what fields are required and what fields are mutually exclusive with one another. Should also add to the readme the commands to start the UI and main; that is ```python -m CogNative.testUI.UI``` and ```python -m CogNative.main```. At some point we might want to allow an option to enter a txt file as input instead of exclusively having the user manually type in the text box.

Issues: The use saved embedding checkbox does not do anything (the command line argument being passed here is hardcoded as 'y'). When you clone a voice and it outputs the filepath at the end in green once it's done, if you click clone voice again it will not get rid of the green filepath and instead will print the "Cloning Voice..." text on top of it.

File: CogNative/testUI/UI.py (hunk 1 above)
Comment (andrew-fennell) [inline CogNative/testUI/UI.py]:
```suggestion
           '-useExistingEmbed', data["input"]["use_embedding"]
```

This should fix the issue.

**Comment by andrew-fennell** [comment]
> Might want to show what folders the user selected for the various audio inputs in the UI.

I agree! I'll add this.

> Should also add to the readme the commands to start the UI and main

This should have been introduced with #84. I will merge main into this branch.

> At some point we might want to allow an option to enter a txt file as input instead of exclusively having the user manually type in the text box.

I just opened another issue to address this. I think it is a good idea. #86

**Comment by andrew-fennell** [comment]
The biggest error in this is that when the GUI exits, the process finishes running. 

I think this is a significant problem, but it is going to take some effort to fix it.

**Comment by andrew-fennell** [comment]
Updated example with a **.txt** file selected as user input.
image

If the text file or "audio to transcribe" file is too long, it will print 80 characters followed by "...".

**Review by JacobSmith1020 — APPROVED** [review]
We probably don't want to leave the file extension for the output up to the user, should just ask for a name and export as a wav by default. Not a big deal if left the way it is though.

There is also still the small visual issue where the green output at the bottom will be overwritten by successive requests to clone a voice and by error messages that may pop up:

image

image

Other than that, this looks great :)

**Comment by andrew-fennell** [comment]
> There is also still the small visual issue where the green output at the bottom will be overwritten by successive requests to clone a voice and by error messages that may pop up:

Due to the demo timeline closing in, I'll merge this and open issues regarding the other smaller issues. These can be fixed in smaller PRs. This isn't best practice, but this tool will be helpful in quickly generating demo audio.

#89 
#90

**Pull request merged**